dds_web/api/user.py

```diff
@@ -876,34 +876,77 @@ def post(self):
         if not (user_email := json_input.get("email")):
             raise ddserr.DDSArgumentError(message="User email missing.")
 
-        # Check if email is registered to a user
+        # Check if the user exists or has a pending invite
         try:
             existing_user = user_schemas.UserSchema().load({"email": user_email})
+            unanswered_invite = user_schemas.UnansweredInvite().load({"email": user_email})
         except sqlalchemy.exc.OperationalError as err:
             raise ddserr.DatabaseError(message=str(err), alt_message="Unexpected database error.")
 
-        if not existing_user:
+        # If the user doesn't exist and doesn't have a pending invite
+        if not existing_user and not unanswered_invite:
             raise ddserr.NoSuchUserError(
-                f"The user with email '{user_email}' does not have access to the specified project."
-                " Cannot remove non-existent project access."
+                f"The user / invite with email '{user_email}' does not have access to the specified project. "
+                "Cannot remove non-existent project access."
             )
 
-        user_in_project = False
-        for user_association in project.researchusers:
-            if user_association.user_id == existing_user.username:
-                user_in_project = True
-                db.session.delete(user_association)
-                project_user_key = models.ProjectUserKeys.query.filter_by(
-                    project_id=project.id, user_id=existing_user.username
-                ).first()
-                if project_user_key:
-                    db.session.delete(project_user_key)
-
-        if not user_in_project:
-            raise ddserr.NoSuchUserError(
-                f"The user with email '{user_email}' does not have access to the specified project."
-                " Cannot remove non-existent project access."
-            )
+        if unanswered_invite:
+            # If there is a unit_id value, it means the invite was associated to a unit
+            # i.e The invite is for a Unit Personel which shouldn't be removed from individual projects
+            if unanswered_invite.unit_id:
+                raise ddserr.UserDeletionError(
+                    "Cannot remove a Unit Admin / Unit User from individual projects"
+                )
+
+            invite_id = unanswered_invite.id
+
+            # Check if the unanswered invite is associated with the project
+            project_invite_key = models.ProjectInviteKeys.query.filter_by(
+                invite_id=invite_id, project_id=project.id
+            ).one_or_none()
+
+            if project_invite_key:
+                # Remove the association if it exists
+                db.session.delete(project_invite_key)
```

Comment:
Add message here instead
```suggestion
            if project_invite_key:
                msg = f"Invited user is no longer associated with {project.public_id}."
                # Remove the association if it exists
                db.session.delete(project_invite_key)
```